src/Form/RedirectSettingsForm.php

```diff
@@ -64,6 +64,12 @@ public function buildForm(array $form, FormStateInterface $form_state) {
     //   '#description' => $this->t('This feature requires <a href="@performance">Cache pages for anonymous users</a> to be enabled and the %variable variable to be TRUE.', array('@performance' => url('admin/config/development/performance'), '%variable' => "\$conf['page_cache_invoke_hooks']")),
     //   '#disabled' => !$cache_enabled || !$invoke_hooks,
     // );
+    $form['redirect_disable_automatic_purge'] = [
+      '#type' => 'checkbox',
+      '#title' => $this->t('Disable automatic purging?'),
+      '#description' => $this->t('Disabling automatic purging will also disable redirect-usage tracking.'),
+      '#default_value' => $config->get('disable_automatic_purge'),
+    ];
     $invoke_hooks = \Drupal::config('system.performance')->get('cache.page.invoke_hooks');
```

Comment:
Nice, can we try to clean up this form a bit more? $invoke_hooks doesn't exist anymore, that's definitely dead code. And if page_cache is enabled, should we force disable this option? I'm not sure that works well, because we'd need the same checks at runtime. Also, that logic will likely move into a n optional module, so we can't easily check anymore if it is enabled or not. So we'd need to go back to a simple description, which we can IMHO also already do now. And then there's the caching setting, it is possible that redirects already get cached in 8.x by default and we'd actually have to do something to disable that.